### PR TITLE
fix(dashboard): key keep-alive PTY on resume+profile, not just tab token (#61284)

### LINKED ISSUE #61284: [Bug]: Dashboard chat sessions fail to render due to recent WebSocket regression
### Bug Description

Dashboard "Silent WebSocket" on Session Switch (v2026.7.7.2+)

## Symptom

When switching chat sessions in the Hermes dashboard, the PTY terminal opens but displays no output and the UI never updates. REST metadata fetching (`api.getSessionDetail`) succeeds. Browser console shows:

```
[chat] PTY WebSocket closed code=1005 reason=
ws closed peer=<ip> reason=client_disconnect(code=1005,reason=) messages=1 parse_errors=0 dispatch_crashes=0 send_failures=0 reaped_sessions=1 detached_sessions=0
```

The issue is that `PTY_REGISTRY.attach_or_spawn()` creates a new Node.js process that takes 1-3 seconds to boot, but the browser WebSocket has no mechanism to wait for output before considering the connection "established." The browser disconnects with `code=1005` before any PTY data arrives.

## Impact

- Session switching in the dashboard is non-functional
- Dashboard PTY shows blank terminal after session switch
- Active session (no switch) works fine — only the transition is broken
- The `ptyAttachToken(forceFresh=true)` path generates a new UUID on every session switch, creating a fresh PTY each time instead of reattaching

### Steps to Reproduce

1. Deploy Docker image v2026.7.7.2
2. Open dashboard, switch between 3+ chat sessions
3. All chat sessions will just re-render the same current session
4. Watch for `code=1005` `messages=1` in agent.log — should transition to `messages=N` (N > 1) after fix
5. Verify PTY renders output after each session switch within 5 seconds

### Expected Behavior

View should change to the selected chat session.

### Actual Behavior

View redraws the current or new chat session instead of the selected session.

### Affected Component

Other

### Messaging Platform (if gateway-related)

_No response_

### Debug Report

```shell
N/A - Fresh install of the docker image.
```

### Operating System

Dockerfile

### Python Version

_No response_

### Hermes Version

_No response_

### Additional Logs / Traceback (optional)

```shell

```

### Root Cause Analysis (optional)

**PTY Keep-Alive System Race Condition** — 6 new commits between v2026.7.1 and v2026.7.7.2 rewrote `/api/pty` from 1:1 socket↔PTY to registry-based with `?attach=<UUID>` token reattachment.

### The Race

| Step | v2026.7.1 (works) | v2026.7.7.2 (broken) |
|------|-------------------|---------------------|
| 1 | Session switch → old WS closes → PTY killed | `forceFresh=true` → new UUID token generated → old WS **detaches only** (PTY lives) |
| 2 | New WS opens → new PTY spawns immediately | `PTY_REGISTRY.attach_or_spawn()` creates new PTY process |
| 3 | Node.js boots, agent initializes (~1-3s) | Same — but browser disconnects **before** output arrives |
| 4 | Output flows to browser | Browser sees empty connection → `code=1005`, `messages=1` |

### Why `messages=1` + `code=1005`

The browser WebSocket opens (1 frame = the handshake), then closes before any PTY output arrives. The PTY process exists but hasn't produced output yet — Node.js is still booting, the agent hasn't started, the drain task hasn't begun forwarding.

## Code Changes (v2026.7.1 → v2026.7.7.2)

| Commit | File | Change |
|--------|------|--------|
| `41166bbe` | `hermes_cli/pty_session.py` | `PtySessionRegistry` with capacity + reap |
| `0ecfbc98` | `hermes_cli/pty_session.py` | `RingBuffer` for output replay |
| `e5ac169c` | `hermes_cli/pty_session.py` | `.attach()`/`.detach()` lifecycle, close codes 4409/4410 |
| `e10e4bca` | `hermes_cli/web_server.py` | `?attach=<token>` reattach path (139 additions, 59 deletions) |
| `79f4f78f` | `web/src/pages/ChatPage.tsx` | `ptyAttachToken()` localStorage + auto-reconnect (55 additions) |
| `c3d2be07` | `hermes_cli/web_server.py` | Periodic reaper wired into dashboard lifespan |

@TinkerOfThings 

### Proposed Fix (optional)

1. **Add a ready-check before closing** — wait for the drain task to start forwarding before allowing the browser WS to timeout
2. **Use `ptyAttachToken(forceFresh=false)` during session switch** — keep the same token so the registry reuses the existing PTY instead of spawning a new one
3. **Increase the browser reconnect backoff** — give Node.js time to boot before the browser gives up and disconnects
4. **Add a `/api/pty/ready` endpoint** — frontend polls this before considering the connection "established"

### Are you willing to submit a PR for this?

- [ ] I'd like to fix this myself and submit a PR

### PR DESCRIPTION
**Pull request by kyssta-exe**
## Summary

Fixes #61284 — Dashboard "Silent WebSocket" on Session Switch.

## Root cause

The keep-alive PTY registry (`PtySessionRegistry`) was keyed only on the per-tab `attach_token`, which is stable across session switches. When the user selects a different chat session:

1. The same `attach_token` is sent
2. `attach_or_spawn()` finds the existing PTY running session A and reattaches
3. The freshly-resolved argv carrying `--resume B` is never spawned
4. The dashboard shows the previous session's terminal

The `code=1005 / messages=1` log line in the issue is benign teardown of the superseded socket, not the root cause.

## Fix

Fold `resume` (which session to resume) and `profile` into the registry key using NUL-byte concatenation:

- Same `attach_token` + same `resume` → reattach (browser refresh preserves keep-alive)
- Same `attach_token` + different `resume` → spawn new PTY (session switch)
- `profile` is also included for the same latent bug with profile switching

```python
registry_key = "\0".join((attach_token, resume or "", profile or ""))
```

The detach call is updated to use the same compound key.

## Verification

1. Open dashboard, switch between 3+ chat sessions — each switch should show the correct session's terminal
2. Refresh the browser while viewing session B — keep-alive reattaches to session B
3. The old session A's PTY lingers to the reaper TTL, so switching back reattaches to A's live agent
4. Bounded by the existing `max_sessions` + reaper

### CODE REVIEW

**Comment by alt-glitch** [comment]
> *This was generated by AI during triage.*

Duplicate of #60745 — both are OPEN, both touch only `hermes_cli/web_server.py`, and both fix the same wrong-session PTY reattach bug (#61284) via the identical mechanism: fold `resume` + `profile` into the PTY registry key as a NUL(`\0`)-joined string so switching sessions spawns a distinct PTY instead of reusing the old one. This PR: `registry_key = "\0".join((attach_token, resume or "", profile or ""))`; #60745: `attach_token = f"{attach_token}\0{profile or ''}\0{resume or ''}"`. Byte-equivalent apart from cosmetics (variable name, profile/resume order). Canonical = earliest OPEN by createdAt: #60745 (2026-07-08) predates this PR (2026-07-09). Related: #61290 (closed same-mechanism backend twin), #61045/#61107 (frontend twins, ChatPage.tsx), #60515 (merged salvage that introduced the keep-alive regression). Maintainer should pick one backend fix.

**Review by teknium1 — COMMENTED** [review]
Thanks for the focused backend fix. The wrong-session premise is present on current main: `hermes_cli/web_server.py:15452-15454` calls the registry with only `attach_token`, and `hermes_cli/pty_session.py:151-153` returns an existing live PTY without spawning the newly resolved argv.

### Problems
- The new key derives from `resume` after the default-chat active-session fallback at `hermes_cli/web_server.py:15405-15406`. A first default-chat connection can use an empty resume key, while a refresh uses the persisted session ID and creates a second PTY rather than reattaching.
- The PR adds no regression coverage beyond current main's same-token test in `tests/test_pty_keepalive_ws.py:7-52`.

### Suggested changes
- Capture the raw query `resume` before fallback and use that raw value, plus `profile`, for the registry identity.
- Add resume-switch, profile-switch, and default-chat-after-fallback regression tests. Open PR #60745 (`1a5680ca`) already demonstrates that shape and coverage.

Automated hermes-sweeper review.

File: hermes_cli/web_server.py [hunk 1]
```diff
@@ -15393,9 +15393,12 @@ def _spawn():
         return
 
     # Keep-alive path: the PTY outlives this socket; reattach by token.
+    # Key on attach_token + resume + profile so switching to a different
```
Comment (teknium1) [inline hermes_cli/web_server.py:15396]: Use the raw query `resume` captured before active-session-file fallback here. On default `/chat`, that fallback can change `resume` from empty to the persisted session ID between refreshes, causing this key to change and the live keep-alive PTY to be replaced.

**Comment by kyssta-exe** [comment]
Stale — no merge activity for 4-6 days. Can resubmit if still needed.
